src/core_plugins/kibana/public/dashboard/directives/grid.js

```diff
@@ -86,7 +72,23 @@ app.directive('dashboardGrid', function ($compile, Notifier) {
           const added = _.difference(panels, currentPanels);
 
           if (removed.length) removed.forEach(removePanel);
-          if (added.length) added.forEach(addPanel);
+          if (added.length) {
+            // See issue https://github.com/elastic/kibana/issues/2138 and the
+            // subsequent fix for why we need to sort here. Short story is that
+            // gridster can fail to render widgets in the correct order, depending
+            // on the specific order of the panels.
+            // See https://github.com/ducksboard/gridster.js/issues/147
+            // for some additional back story.
+            added.sort((a, b) => {
```

Comment:
It was slightly confusing to me why this worked at first, but then I realized it is sorting a *copy* of `state.panels` rather than modifying it. I think future readers would benefit from that fact being a bit more obvious somehow.

Maybe changing `added.forEach(addPanel);` to `addPanels(added)` and making `addPanels()` do something like:

```js
function addPanels(panels) {
  // copy and sort the panels by row before applying them so
  // that gridster lays them out correctly
  const sortedPanels = panels.slice().sort(...);
  sortedPanels.forEach(appPanel);
}
```